File: workshops/2023/BRIDGES-Summer-23-07/openswim/graphIMDB/python_answer/graph_imdb.py
from bridges import *

def main():
    bridges = Bridges(1, "BRIDGES_USER_NAME", "BRIDGES_API_KEY")

    bridges.set_title("UNCC BRIDGES Workshop Number: IMDB Actor-Movie Data")
    bridges.set_description("Building a graph from IMDB actor movie pairs")

    graph = GraphAdjList()

    # retrieve the IMDB actor movie data
    actor_list = get_actor_movie_imdb_data(1814)
    
    # Traverse the list and populate the graph with actors and movies
    for m in range(len(actor_list)):
        actor = actor_list[m].actor
        movie = actor_list[m].movie

        # Add vertices for unique actor/movies
        if graph.get_vertex(actor) == None:
            graph.add_vertex(actor, None)

        if graph.get_vertex(movie) == None:
            graph.add_vertex(movie, None)

        # Add edge between this actor and movie both ways
        graph.add_edge(movie, actor)
        graph.add_edge(actor, movie)

        # If the actors are the target actors highlight the links and vertices
        if actor == 'Cate_Blanchett':
            graph.get_link_visualizer(actor, movie).color = "red"
            graph.get_visualizer(actor).color = "red"
            graph.get_visualizer(actor).size = 49.0 
            graph.get_visualizer(movie).color = "red"
        elif actor == 'Kevin_Bacon_(I)':
            graph.get_link_visualizer(actor, movie).color = "green"
            graph.get_visualizer(actor).size = 49.0 
            graph.get_visualizer(actor).color = "green"
            graph.get_visualizer(movie).color = "green"

    bridges.set_data_structure(graph)
    bridges.visualize()

if __name__ == '__main__':
    main()

File: workshops/2024/bridges/openswim/graphIMDB/c++_answer/graphIMDB.cpp

#include <iostream>
#include <string>
#include <vector>
#include <queue>

using namespace std;

#include "Bridges.h"
#include "GraphAdjList.h"
#include "DataSource.h"
#include "data_src/ActorMovieIMDB.h"

using namespace bridges;

int  main() {

  // Initialize BRIDGES with your credentials
  Bridges bridges(103, "BRIDGES_USER_ID", "BRIDGES_API_KEY");
  
  // set title, description for visualization
  bridges.setTitle("UNCC BRIDGES Workshop: May 2022");
  bridges.setDescription("Drawing a graph using IMDB Actor-Movie Pairs");

  // use an adjacency list based graph
  GraphAdjList<string> graph;

  // get the actor movie IMDB data through the BRIDGES API
  DataSource ds;
  vector<ActorMovieIMDB> actor_list = ds.getActorMovieIMDBData(1814);

  for (int  k = 0; k < actor_list.size(); k++) {
    string actor, movie;

    // get the actor and movie names
    actor = actor_list[k].getActor();
    movie = actor_list[k].getMovie();

    // our graph needs to have a unique set of actors and movies;
    // so create the actor and movie vertices only if they dont already
    // exit; use an STL map to check for that

	if (!graph.getVertex(actor))	// doesnt exist
		graph.addVertex(actor);

	if (!graph.getVertex(movie))	// doesnt exist
		graph.addVertex(movie);

	// add edge
	graph.addEdge (actor, movie);

    // TO DO : Highlight "Cate_Blanchett" node and the movie nodes she is
    // connected to  in "red" and do the same for "Kevin_Bacon_(I)" in "green"
    // specify colors by Color("red"), for example
    if (actor == "Cate_Blanchett") {
      graph.getVisualizer (actor)->setColor (Color("red"));
      graph.getLinkVisualizer (actor, movie)->setColor (Color("red"));
      graph.getVisualizer (movie)->setColor (Color("red"));
    }
    if (actor == "Kevin_Bacon_(I)") {
      graph.getVisualizer (actor)->setColor (Color("green"));
      graph.getLinkVisualizer (actor, movie)->setColor (Color("green"));
      graph.getVisualizer (movie)->setColor (Color("green"));
    }
  }
  bridges.setDataStructure(&graph);
  bridges.visualize();

  return 0;
}
